File: ClimbLog.Api/ClimbLog.Domain/Models/Entities/BetaVideo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ClimbLog.Domain.Models.Entities;

public class BetaVideo
{
    public int Id { get; set; }
    public int RouteId { get; set; }
    public required string Url { get; set; }
    public string? ThumbnailUrl { get; set; }
}

File: climb-log-flutter/lib/climasys/api/models/beta_video.dart
class BetaVideo {
  final int id;
  final int routeId;
  final String url;
  final String thumbnailUrl;

  BetaVideo({
    required this.id,
    required this.routeId,
    required this.url,
    required this.thumbnailUrl
  });

  factory BetaVideo.fromJson(Map<String, dynamic> json) {
    return BetaVideo(
        id: json['id'],
        routeId: json['routeId'],
        url: json['url'],
        thumbnailUrl: json['thumbnailUrl']
    );
  }
}
